### PR TITLE
Add Example Terraform file

### PR DESCRIPTION
**Pull request by DorukAkinci**

### CODE REVIEW

**Comment by DorukAkinci** [comment]
@JohannesRudolph @felixzieger, Could you guys check this terraform demo template. I do not have any access, so i improvised to build one 👍  If you guys need any help also let me know.

meshLove ❤️

File: demo.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+provider "meshapi" {
+    url = "https://DEMO.meshstack.io"
+    headers = {
+        Authorization = "BASE64_VAL"
```
Comment (felixzieger) [inline demo.tf]:
```suggestion
        Authorization = "Basic ..." # The three dots must be replaced with the BASE64 decoded value of `username:password` (without the ticks)
```

File: demo.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
+provider "meshapi" {
+    url = "https://DEMO.meshstack.io"
```
Comment (felixzieger) [inline demo.tf]:
```suggestion
    url = "..." # Example: url = "https://federation.demo.meshstack.io"
```

File: demo.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            ProjectContact = "DeveloperUser"
+            CostCenter = 0000
+        }
+    )
+}
+
+// GET DEMO USER'S DATA TO SHOW DATA USAGE
+data "meshapi_mesh_user" "demo" {
+    name = "DEMO@meshcloud.io"
```
Comment (felixzieger) [inline demo.tf]:
```suggestion
    name = "..." # Example: name = "demo-user@meshcloud.io"
```

**Review by felixzieger — APPROVED** [review]
Hi Doruk! Thanks for submitting the example 👍 Just a few doc suggestions, the example itself works for me

**Comment by DorukAkinci** [comment]
Hello @felixzieger 👋 
I have created a small list of examples to develop this API beforehand and to understand how it works. Probably you can access the list on the following page as well. 
https://github.com/stars/DorukAkinci/lists/terraform-providers

Also when it comes to terraform providers, there is a trick to find all of them :) you can quickly find them in github if you search `terraform-provider-` prefix. This is a requirement from Hashicorp itself.

The best documentation is that Hashicorp's own plugin development documentation. There is no other valuable guide that i could see. Once you figure out how Terraform Schemas works, the other part is pure Go development. 
https://www.terraform.io/plugin/sdkv2/schemas

**Comment by DorukAkinci** [comment]
Meanwhile i have no write access on this Repo, so you should merge it 😄  I have added one more commit to fix tagging.

**Comment by JohannesRudolph** [comment]
hey @DorukAkinci thanks for adding the example and the additional tips, will merge this now. 

We have great plans for this provider going forward together with our new https://github.com/meshcloud/landing-zone-construction-kit, it will be awesome! Thank you so much for kicking this off.

**Pull request merged**